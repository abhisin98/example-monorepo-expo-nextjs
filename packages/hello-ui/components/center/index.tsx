import type { VariantProps } from "@gluestack-ui/utils/nativewind-utils";
import React from "react";
import { View, ViewProps } from "react-native";

import { centerStyle } from "./styles";

type ICenterProps = ViewProps & VariantProps<typeof centerStyle>;

const Center = React.forwardRef<React.ComponentRef<typeof View>, ICenterProps>(function Center({ className, ...props }, ref) {
  return <View className={centerStyle({ class: className })} {...props} ref={ref} />;
});

Center.displayName = "Center";

export { Center };
