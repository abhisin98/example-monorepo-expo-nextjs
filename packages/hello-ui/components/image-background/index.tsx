"use client";
import { tva } from "@gluestack-ui/utils/nativewind-utils";
import React from "react";
import { ImageBackground as RNImageBackground } from "react-native";

const imageBackgroundStyle = tva({});

const ImageBackground = React.forwardRef<React.ComponentRef<typeof RNImageBackground>, React.ComponentProps<typeof RNImageBackground>>(function ImageBackground({ className, ...props }, ref) {
  return (
    <RNImageBackground
      className={imageBackgroundStyle({
        class: className,
      })}
      {...props}
      ref={ref}
    />
  );
});

ImageBackground.displayName = "ImageBackground";

export { ImageBackground };
