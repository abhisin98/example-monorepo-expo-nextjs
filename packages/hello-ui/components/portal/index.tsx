"use client";
import { Overlay } from "@gluestack-ui/core/overlay/creator";
import { cssInterop } from "nativewind";
import React from "react";

cssInterop(Overlay, { className: "style" });

const Portal = React.forwardRef<React.ComponentRef<typeof Overlay>, React.ComponentProps<typeof Overlay>>(function Portal({ ...props }, ref) {
  return <Overlay {...props} ref={ref} />;
});

Portal.displayName = "Portal";

export { Portal };
