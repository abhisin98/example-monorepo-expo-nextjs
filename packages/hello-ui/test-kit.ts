import { render as rn, RenderOptions, renderCurrentTest as rnct, RenderCurrentTestOptions } from "nativewind/test";
import React from "react";

import tailwindRootConfig from "./tailwind.config.js";
export * from "nativewind/test";

const css = `
@tailwind base;
@tailwind components;
@tailwind utilities;
`;

export const render = (component: React.ReactElement<any>, options?: RenderOptions) => {
  return rn(component, {
    config: tailwindRootConfig,
    css,
    ...options,
  });
};

export const renderCurrentTest = (options?: RenderCurrentTestOptions) => {
  return rnct({
    config: tailwindRootConfig,
    css,
    ...options,
  });
};
