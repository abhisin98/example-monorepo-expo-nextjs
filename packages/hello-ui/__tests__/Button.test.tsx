import { cleanup, fireEvent } from "@testing-library/react-native";

import { Button, ButtonText, ButtonIcon, ButtonSpinner, ButtonGroup } from "../components/button";
import { GluestackUIProvider } from "../components/gluestack-ui-provider";
import { AddIcon } from "../components/icon";
import { render } from "../test-kit";

afterAll(cleanup);

//-----------------------------------------------------------------

describe("Button", () => {
  it("renders Button with text", async () => {
    const element = await render(
      <GluestackUIProvider mode='light'>
        <Button>
          <ButtonText>Click me</ButtonText>
        </Button>
      </GluestackUIProvider>
    );
    expect(element.getByText("Click me")).toBeTruthy();
    expect(element).toMatchSnapshot();
  });

  it("calls onPress when pressed", async () => {
    const onPressMock = jest.fn();
    const element = await render(
      <Button onPress={onPressMock}>
        <ButtonText>Press</ButtonText>
      </Button>
    );
    fireEvent.press(element.getByRole("button"));
    expect(onPressMock).toHaveBeenCalled();
    expect(element).toMatchSnapshot();
  });

  it("renders ButtonIcon", async () => {
    const element = await render(
      <Button>
        <ButtonIcon as={AddIcon} testID='icon' />
        <ButtonText>Star</ButtonText>
      </Button>
    );
    expect(element.getByTestId("icon")).toBeTruthy();
    expect(element).toMatchSnapshot();
  });

  it("renders ButtonSpinner when loading", async () => {
    const element = await render(
      <Button>
        <ButtonSpinner testID='spinner' />
        <ButtonText>Loading</ButtonText>
      </Button>
    );
    expect(element.getByTestId("spinner")).toBeTruthy();
    expect(element).toMatchSnapshot();
  });

  it("renders ButtonGroup with multiple buttons", async () => {
    const element = await render(
      <ButtonGroup>
        <Button>
          <ButtonText>One</ButtonText>
        </Button>
        <Button>
          <ButtonText>Two</ButtonText>
        </Button>
      </ButtonGroup>
    );
    expect(element.getAllByRole("button")).toHaveLength(2);
    expect(element).toMatchSnapshot();
  });
});
