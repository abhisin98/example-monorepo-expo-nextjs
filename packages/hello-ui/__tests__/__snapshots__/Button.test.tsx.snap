// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button calls onPress when pressed 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  dataSet={
    {
      "active": "false",
      "disabled": "false",
      "focus": "false",
      "focusVisible": "false",
      "hover": "false",
    }
  }
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="button"
  states={
    {
      "active": false,
      "disabled": undefined,
      "focus": false,
      "focusVisible": false,
      "hover": false,
    }
  }
  style={
    {
      "backgroundColor": undefined,
      "borderColor": undefined,
    }
  }
>
  <Text
    dataSet={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    style={
      {
        "color": undefined,
      }
    }
  >
    Press
  </Text>
</View>
`;

exports[`Button renders Button with text 1`] = `
<View
  style={
    {
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    style={
      {
        "backgroundColor": "rgba(51, 51, 51, 1)",
        "borderColor": "rgba(102, 102, 102, 1)",
      }
    }
  >
    <Text
      dataSet={
        {
          "active": false,
          "disabled": undefined,
          "focus": false,
          "focusVisible": false,
          "hover": false,
        }
      }
      states={
        {
          "active": false,
          "disabled": undefined,
          "focus": false,
          "focusVisible": false,
          "hover": false,
        }
      }
      style={
        {
          "color": "rgba(254, 254, 255, 1)",
        }
      }
    >
      Click me
    </Text>
  </View>
</View>
`;

exports[`Button renders ButtonGroup with multiple buttons 1`] = `
<View
  flexDirection="row"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    style={
      {
        "backgroundColor": undefined,
        "borderColor": undefined,
      }
    }
  >
    <Text
      dataSet={
        {
          "active": false,
          "disabled": undefined,
          "focus": false,
          "focusVisible": false,
          "hover": false,
        }
      }
      states={
        {
          "active": false,
          "disabled": undefined,
          "focus": false,
          "focusVisible": false,
          "hover": false,
        }
      }
      style={
        {
          "color": undefined,
        }
      }
    >
      One
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    style={
      {
        "backgroundColor": undefined,
        "borderColor": undefined,
      }
    }
  >
    <Text
      dataSet={
        {
          "active": false,
          "disabled": undefined,
          "focus": false,
          "focusVisible": false,
          "hover": false,
        }
      }
      states={
        {
          "active": false,
          "disabled": undefined,
          "focus": false,
          "focusVisible": false,
          "hover": false,
        }
      }
      style={
        {
          "color": undefined,
        }
      }
    >
      Two
    </Text>
  </View>
</View>
`;

exports[`Button renders ButtonIcon 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  dataSet={
    {
      "active": "false",
      "disabled": "false",
      "focus": "false",
      "focusVisible": "false",
      "hover": "false",
    }
  }
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="button"
  states={
    {
      "active": false,
      "disabled": undefined,
      "focus": false,
      "focusVisible": false,
      "hover": false,
    }
  }
  style={
    {
      "backgroundColor": undefined,
      "borderColor": undefined,
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100%"
    bbWidth="100%"
    fill="none"
    focusable={false}
    meetOrSlice={0}
    minX={0}
    minY={0}
    role="img"
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    stroke="currentColor"
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {},
        {
          "flex": 0,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="icon"
    vbHeight={24}
    vbWidth={24}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "type": 2,
        }
      }
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M12 5V19"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
        <RNSVGPath
          d="M5 12H19"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    dataSet={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    style={
      {
        "color": undefined,
      }
    }
  >
    Star
  </Text>
</View>
`;

exports[`Button renders ButtonSpinner when loading 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  dataSet={
    {
      "active": "false",
      "disabled": "false",
      "focus": "false",
      "focusVisible": "false",
      "hover": "false",
    }
  }
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="button"
  states={
    {
      "active": false,
      "disabled": undefined,
      "focus": false,
      "focusVisible": false,
      "hover": false,
    }
  }
  style={
    {
      "backgroundColor": undefined,
      "borderColor": undefined,
    }
  }
>
  <ActivityIndicator
    aria-label="loading"
    dataSet={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    tabIndex={0}
    testID="spinner"
  />
  <Text
    dataSet={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    style={
      {
        "color": undefined,
      }
    }
  >
    Loading
  </Text>
</View>
`;
